experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -279,15 +277,12 @@ export const getSequenceField = <TTypes extends AllowedTypes>(list: TreeListNode
 // typed data prior to forwarding to 'LazySequence.insert*()'.
 function contextualizeInsertedListContent(
 	insertedAtIndex: number,
-	content: (
-		| InsertableTreeNodeUnion<AllowedTypes>
-		| IterableTreeListContent<InsertableTreeNodeUnion<AllowedTypes>>
-	)[],
-): ExtractedFactoryContent<ContextuallyTypedNodeData[]> {
+	content: readonly (InsertableContent | IterableTreeListContent<InsertableContent>)[],
+): ExtractedFactoryContent {
 	return extractFactoryContent(
-		content.flatMap((c) =>
-			c instanceof IterableTreeListContent ? Array.from(c) : c,
-		) as ContextuallyTypedNodeData[],
+		content.flatMap((c): InsertableContent[] =>
+			c instanceof IterableTreeListContent ? Array.from(c) : [c],
```

Comment:
If the content being inserted into the List is another list, it should be inserted as a child, not flattened. This change fixes that edge case.